gulp-tasks/project.js

```diff
@@ -0,0 +1,138 @@
+function splitDependencies() {
+  return project.dependencies().pipe(project.splitHtml());
+}
+
+// Returns a WriteableStream to rejoin all split files
+function rejoin() {
+  return project.rejoinHtml();
+}
+
+// Returns a function which accepts refernces to functions that generate
+// ReadableStreams. These ReadableStreams will then be merged, and used to
+// generate the bundled and unbundled versions of the site.
+// Takes an argument for the user to specify the kind of output they want
+// either bundled or unbundled. If this argument is omitted it will output both
+function merge(source, dependencies) {
+  return function output() {
+    const mergedFiles = mergeStream(source(), dependencies())
+      .pipe(project.analyzer);
+    const bundleType = global.config.build.bundleType;
+    let outputs = [];
+
+    if (bundleType === 'both' || bundleType === 'bundled') {
+      outputs.push(writeBundledOutput(polymer.forkStream(mergedFiles)));
+    }
+    if (bundleType === 'both' || bundleType === 'unbundled') {
+      outputs.push(writeUnbundledOutput(polymer.forkStream(mergedFiles)));
+    }
+
+    return Promise.all(outputs);
+  };
+}
+
+// Run the files through a bundling step which will vulcanize/shard them
+// then output to the dest dir
+function writeBundledOutput(stream) {
+  return new Promise(resolve => {
+    stream.pipe(project.bundler)
+      .pipe(gulp.dest(bundledPath))
+      .on('end', resolve);
+  });
+}
+
+// Just output files to the dest dir without bundling. This is for projects that
+// use HTTP/2 server push
+function writeUnbundledOutput(stream) {
+  return new Promise(resolve => {
+    stream.pipe(gulp.dest(unbundledPath))
+      .on('end', resolve);
+  });
+}
+
+// Returns a function which takes an argument for the user to specify the kind
+// of bundle they're outputting (either bundled or unbundled) and generates a
+// service worker for that bundle.
+// If this argument is omitted it will create service workers for both bundled
+// and unbundled output
+function serviceWorker() {
+  const bundleType = global.config.build.bundleType;
+  let workers = [];
+
+  if (bundleType === 'both' || bundleType === 'bundled') {
+    workers.push(writeBundledServiceWorker());
+  }
+  if (bundleType === 'both' || bundleType === 'unbundled') {
+    workers.push(writeUnbundledServiceWorker());
+  }
+
+  return Promise.all(workers);
+}
+
+// Returns a Promise to generate a service worker for bundled output
+function writeBundledServiceWorker() {
+  return polymer.addServiceWorker({
+    project: project,
+    buildRoot: bundledPath,
+    swConfig: global.config.swPrecacheConfig,
+    serviceWorkerPath: global.config.serviceWorkerPath,
+    bundled: true
+  });
+}
+
+// Returns a Promise to generate a service worker for unbundled output
+function writeUnbundledServiceWorker() {
+  return polymer.addServiceWorker({
+    project: project,
+    buildRoot: unbundledPath,
+    swConfig: global.config.swPrecacheConfig,
```

Comment:
Please also incorporate the changes of https://github.com/AreaFiftyLAN/lancie-frontend/pull/393